src/components/SuperDrawer/useSuperDrawer.tsx

```diff
@@ -151,12 +163,17 @@ export function useSuperDrawer(): UseSuperDrawerHook {
         // When opening a new SuperDrawer, check if we can close the previous
         // SuperDrawer content. Bail if it fails
         // TODO: There needs to be try again logic here...
-        if (!closeActions.closeDrawer()) {
-          return false;
+        debugger;
+        const maybeValidated = validateDrawerChange(openNewModal);
+        if (maybeValidated === false) {
+          return maybeValidated
         }
+        function openNewModal () {
```

Comment:
Maybe rename to `doOpen()`? Or something without "modal" in it, since that term is also used for `useModal` things.

...or actually maybe even just make it a lambda, i.e. maybe this becomes:

```
openInDrawer(opts) {
  validateDrawerChange(() => {
    contentStack.current = [{ kind: "open", opts }];
  })
};
```

Reply:
Good call on the lambda since we don't need that extra `openNewModal` call now.